tests/mypy_test.py

```diff
@@ -341,6 +341,9 @@ def test_third_party_distribution(distribution: str, args: TestConfig) -> TestRe
     seen_dists: set[str] = set()
     add_third_party_files(distribution, files, args, configurations, seen_dists)
 
+    if not files and args.filter:
+        return TestResults(0, 0)
+
```

Comment:
previously on line 351 of this file, it would sys.exit after searching the very first package and finding out that none of the files match.

Now it early-skips all the non-matching packages when we're filtering.

Reply:
There might be a different or better way to solve this, but this definitely works.